libethereum/Executive.cpp

```diff
@@ -331,12 +353,25 @@ bool Executive::create(Address _sender, u256 _endowment, u256 _gasPrice, u256 _g
 	// Schedule _init execution if not empty.
 	if (!_init.empty())
 		m_ext = make_shared<ExtVM>(m_s, m_envInfo, m_sealEngine, m_newAddress, _sender, _origin, _endowment, _gasPrice, bytesConstRef(), _init, sha3(_init), m_depth);
-	else if (m_s.addressHasCode(m_newAddress))
-		// Overwrite with empty code in case the account already has a code
-		// (address collision -- not real life case but we can check it with
-		// synthetic tests).
-		m_s.setCode(m_newAddress, {});
-
+	
+	if (m_envInfo.number() < m_sealEngine.chainParams().u256Param("metropolisForkBlock"))
+	{
+		if (m_s.addressHasCode(m_newAddress))
+			// Overwrite with empty code in case the account already has a code
+			// (address collision -- not real life case but we can check it with
+			// synthetic tests).
+			m_s.setCode(m_newAddress, {});
+	}
+	else
+	{
+		if (accountAlreadyExist)
+		{
+			clog(StateSafeExceptions) << "Address already used: " << m_newAddress;
+			m_gas = 0;
+			m_excepted = TransactionException::AddressAlreadyUsed;
+			revert();
```

Comment:
In this case, I think true must be returned from this function, indicating that the VM execution is not necessary.